run/system-package/main.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from flask import Flask, make_response, request
+import os
+import subprocess
+import sys
+
+app = Flask(__name__)
+
+# [START run_system_package_handler]
+@app.route('/diagram.png', methods=['GET'])
+def index():
+    try:
+        image = create_diagram(request.args.get('dot'))
+
+        response = make_response(image)
+        response.headers.set('Content-Type', 'image/png')
+        return response
+
+    except Exception as e:
+        print(f'error: {e}')
+
+        # Flush the stdout to avoid log buffering.
+        sys.stdout.flush()
+
+        if 'syntax' in str(e):
+            return f'Bad Request: {e}', 400
+
+        return 'Internal Server Error', 500
+# [END run_system_package_handler]
+
+
+# [START run_system_package_exec]
+def create_diagram(dot):
+    if not dot:
+        raise Exception('syntax: no graphviz definition provided')
+
+    dot_args = ['/usr/bin/dot',  # Command to run
```

Comment:
Where is `/usr/bin/dot` coming from? It's not on my machine. Perhaps add a comment around this.

Reply:
LMK if this is clearer!

Reply:
I am still seeing `bash: /usr/bin/dot: No such file or directory`.

I think it would be important to state the dependency of this snippet. Maybe look at the other language samples.



```py
# Uses local `dot` binary from Graphviz: https://graphviz.gitlab.io 
dot_args = ['/usr/bin/dot',
  ...
```

Reply:
This is now at the top of the file with a more verbose explanation of where it comes from, how to install locally, and how to find the filepath.